packages/select/src/select.tsx

```diff
@@ -26,18 +26,18 @@ import type {
   SelectItemProps,
   SelectOverlayProps,
   SelectPortalProps,
-  SelectRootProps as ISelectRootProps,
+  SelectRootProps,
   SelectSeparatorProps,
   SelectValueProps,
   Option,
 } from './types';
 
-interface SelectRootProps extends Partial<ISelectRootProps> {
-  defaultOpen?: boolean;
-  defaultValue?: Option;
-}
-
-interface IRootContext extends ISelectRootProps {
+interface IRootContext extends SelectRootProps {
+  value: Option;
+  onValueChange: (option: Option) => void;
+  open: boolean;
+  onOpenChange: (value: boolean) => void;
+  disabled?: boolean;
```

Comment:
i didn't mean to extend, i wanted write it this way: 
```ts
interface IRootContext {
  value: Option;
  onValueChange: (option: Option) => void;
  open: boolean;
  onOpenChange: (value: boolean) => void;
  disabled?: boolean;
  triggerPosition: LayoutPosition | null;
  setTriggerPosition: (triggerPosition: LayoutPosition | null) => void;
  contentLayout: LayoutRectangle | null;
  setContentLayout: (contentLayout: LayoutRectangle | null) => void;
  nativeID: string;
}
```
since after a `Required<SelectRootProps>` i would also omit a bunch of values

Reply:
Oh ok I see. No change needed here